packages/next-on-pages/templates/_worker.js/routes-matcher.ts

```diff
@@ -189,8 +189,23 @@ export class RoutesMatcher {
 
 		const rewriteKey = 'x-middleware-rewrite';
 		const rewriteHeader = resp.headers.get(rewriteKey);
+
 		if (rewriteHeader) {
 			const newUrl = new URL(rewriteHeader, this.url);
+
+			const thisHost = this.url.hostname;
+			const rewriteHost = newUrl.hostname;
+			const rewriteIsExternal = thisHost !== rewriteHost;
+
+			if (rewriteIsExternal) {
+				const externalResp = await fetch(newUrl, this.reqCtx.request);
+				// The rewrite is external so we do not want to continue and we just
+				// set the final response body, status and headers here.
+				this.body = externalResp.body;
+				this.status = externalResp.status;
+				applyHeaders(this.headers.normal, externalResp.headers);
```

Comment:
I've just been looking at the part where we generate a response and I wonder if we should just update the path to be the full URL here instead of doing the fetch etc., and let that be handled in the generate response function instead.

https://github.com/cloudflare/next-on-pages/blob/1e73555dcb2ead5a83436ed619d5eec660b6c2f3/packages/next-on-pages/templates/_worker.js/handleRequest.ts#L99-L103

What do you think?

Reply:
Yes, of course! I totally missed that :facepalm: